artemis/modules/joomla_scanner.py

```diff
@@ -34,15 +35,33 @@ def _is_version_old(self, version: str, age_threshold_days: int = 180) -> bool:
             assert cache_result
             data = json.loads(cache_result)
 
+        version_parsed = semver.VersionInfo.parse(version)
+        if version_parsed.major < 3:
+            return True
+
+        is_current_version_old = True  # If we don't find the version on releases list, that means, it's old
         for release in data:
             if release["tag_name"] == version:
-                return (
+                is_current_version_old = (
                     datetime.datetime.utcnow().replace(tzinfo=pytz.utc)
                     - datetime.datetime.fromisoformat(release["published_at"])
                 ).days > age_threshold_days
 
-        # Cannot check version age
-        return True
+        is_newer_version_available = False
+        for release in data:
+            release_version = release["tag_name"]
+            release_version_parsed = semver.VersionInfo.parse(release_version)
+            if (
+                release_version_parsed.major == version_parsed.major
+                and release_version_parsed.compare(version_parsed) > 0
+            ):
+                is_newer_version_available = True
```

Comment:
rewrite as well.
```py
if (condition):
     variable = True
```
is not the best practice. Also `release_version` is 15 characters, while `release["tag_name"]` is 19, not really much of a helper. it would be much better to split if into 2 helper variables (since they are long) and remove `release_version` entirely.

Also, `release_version_parsed.compare(version_parsed) > 0` needs a comment. How do you compare 2 semver and result is an int???